executor/adapter.go

```diff
@@ -1337,17 +1337,48 @@ func (a *ExecStmt) GetTextToLog() string {
 	return sql
 }
 
-func (a *ExecStmt) observeStmtBeginForTopSQL() {
+func (a *ExecStmt) observeStmtBeginForTopSQL(ctx context.Context) context.Context {
```

Comment:
Note (no need to change in this PR): a better API design should be a hook framework for statement begin and statement finish events, instead of something xxxForTopSQL. (The observer pattern)

cc @mornyx @zhongzc